### PR TITLE
1880 Refactored Dropdown components

### LINKED ISSUE #1880: Split DropdownButton component in two : DropdownButton and DropdownMenu
## Scope & Context

Split DropdownButton component in two, because in some cases we need DropDownMenu without DropdownButton.

## Technical inputs

- Create two components : DropdownButton and DropdownMenu
- DropdownButton will import DropdownMenu
- These two components should export hooks : useDropdownButton() and useDropdownMenu()

### PR DESCRIPTION
**Pull request by bosiraphael**
Created multiple dropdown components instead of the DropdownButton : 
- DropdownMenu
- ColumnHeadWithDropdown
- ViewBarDropdownButton

You can now pass a clickable component as a prop to DropDownMenu, and also an offset.

### CODE REVIEW

File: front/src/modules/ui/view-bar/components/ViewBarDropdownButton.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Keys } from 'react-hotkeys-hook';
+import { Placement } from '@floating-ui/react';
+
+import { DropdownMenu } from '@/ui/dropdown/components/DropdownMenu';
+import { HotkeyScope } from '@/ui/utilities/hotkey/types/HotkeyScope';
+
+type DropdownButtonProps = {
+  buttonComponents?: JSX.Element | JSX.Element[];
```
Comment (charlesBochet) [inline front/src/modules/ui/view-bar/components/ViewBarDropdownButton.tsx:8]: buttonComponent:

**Review by charlesBochet — APPROVED** [review]

File: front/src/modules/ui/dropdown/components/DropdownMenu.tsx [hunk 2]
```diff
@@ -74,25 +75,38 @@ export const DropdownButton = ({
...
     () => {
-      closeDropdownButton();
+      closeDropdown();
     },
     dropdownHotkeyScope.scope,
-    [closeDropdownButton],
+    [closeDropdown],
   );
-
+  console.log(floatingStyles);
```
Comment (lucasbordeau) [inline front/src/modules/ui/dropdown/components/DropdownMenu.tsx:83]: console.log

File: front/src/modules/ui/dropdown/components/DropdownMenu.tsx [hunk 3]
```diff
@@ -23,45 +23,46 @@ type OwnProps = {
...
-    useDropdownButton({
-      dropdownId,
-    });
+  const { isDropdownOpen, toggleDropdown, closeDropdown } = useDropdown({
+    dropdownId,
+  });
 
   const { refs, floatingStyles } = useFloating({
     placement: dropdownPlacement,
     middleware: [flip(), offset({ mainAxis: 8, crossAxis: 0 })],
```
Comment (lucasbordeau) [inline front/src/modules/ui/dropdown/components/DropdownMenu.tsx:52]: Can we place offset in useFloating which already handles all of this.

**Review by charlesBochet — APPROVED** [review]
LGTM

**Pull request merged**